src/howTos/dev/services.md

```diff
@@ -0,0 +1,112 @@
+
+In contrast to konnectors, services have the same permissions as the web application and are not intended to collect information from the outside. It is rather meant to asynchronously analyse data inside the cozy and emit some output once the task is done. However, they share the same mechanisms as the konnectors to describe how and when they should be executed: via our [trigger system](https://github.com/cozy/cozy-stack/blob/master/docs/jobs.md).
+
+
+ background and offline process to analyse the user's data and emit some output even without the user being on the application. These part of the application are called services and can be declared as part of the application in its manifest.
+
+
+
+## Example
+
+You can find an example of an existing service in the [cozy-banks app](https://github.com/cozy/cozy-banks/blob/master/src/targets/services/onOperationCreate.js).
+
+## Service declaration
+
+The service must be declared in the app [manifest](https://docs.cozy.io/en/dev/app/#read-the-manifest). For example:
+
+```json
+"services": {
+  "onOperationCreate": {
+    "type": "node",
+    "file": "onOperationCreate.js",
+    "trigger": "@event io.cozy.bank.operations:CREATED",
+    "debounce": "3m"
+  }
+}
+```
+
+Here is an explanation of the fields:
+
+* `type`: describe the code type (only `node` for now).
+* `file`: the single (packaged) file containing the code to execute as a service.
+* `trigger`: what triggers the service. It must follow the available triggers described in the [jobs documentation]( https://github.com/cozy/cozy-stack/blob/master/docs/jobs.md). In this example, the trigger is a bank operation creation.
+* `debounce` (optionnal): force a minimal delay between two runs of the service. `m` is for minutes and `s` for seconds. if this parameter is omitted, the service will be executed as soon as it can.
+
+## Build
+
+The service must be packaged into a single file containing all the dependencies. An example of a webpack rule is available [here](https://github.com/cozy/cozy-banks/blob/master/config/webpack.target.services.js). Note that `target: 'node'` is important as the service is run as a Node.js process.
+
+A `watch:services` entry can also be added in the package.json, like in this [example](https://github.com/cozy/cozy-banks/blob/master/package.json). Then the service is simply built with this command:
+
+```bash
+yarn watch:services
+```
+
+### Stack
+
+As the service is run on a dedicated process on the server side, a running stack is necessary. For development, it is easier to use a stack installed without docker, as some configuration is necessary, notably to retrieve the logs of the service. For the stack's installation instructions, follow the [guide](https://github.com/cozy/cozy-stack/blob/master/docs/INSTALL.md).
+
```

Comment:
You mean the context? We can now pass a config file to the docker if we need [some specific configuration now](https://docs.cozy.io/en/howTos/dev/runCozyDocker/#run-with-a-custom-stack-config-file)